SilKit/ci/check_formatting.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: 2024 Vector Informatik GmbH
+#
+# SPDX-License-Identifier: MIT
+from pathlib import Path
+from shutil import which
+import re
+import subprocess
+import sys
+
+## Convenience
+def log(fmt, *args):
+    print(fmt.format(*args))
+
+def info(fmt, *args):
+    log("INFO: " + fmt, *args)
+
+def warn(fmt, *args):
+    log("::warning:: " + fmt, *args)
+
+def die(status, fmt, *args):
+    log("ERROR: " + fmt, *args)
+    sys.exit(status)
+
+## Check if clang-format is installed
+
+if which('clang-format') is None:
+    warn("No clang-format found!")
+    die(1, "Please install clang-format!")
+
+# Check for supported clang-format version
+format_version = subprocess.run(['clang-format', '--version'], capture_output=True)
+
+version_reg = re.compile('^.* clang-format version (\d+)\.(\d+)\.(\d+).*')
+version = re.match(version_reg, format_version.stdout.decode('utf-8'))
+
+if version is None or len(version.groups()) != 3:
+    die(2, "ERROR: Could not get the clang-format version!")
+
+major, minor, patch = version.group(1,2,3)
+if int(major) < 13:
+    die(3, "clang{} not supported!\r\n       Minimum supported version is clang14!", major)
+
+info("clang-format-{}.{}.{} found!", major, minor, patch)
+
+############################################################################################################################
+fileExtensions = ["*.cpp", "*.ipp", "*.c", "*.hpp", "*.h"]
+dirs = ["SilKit", "Demos", "Utilities"]
+
+formatting_correct = True
+
+# Open LogFile to get some data on our poorly formatted files!
+with open('FormatLog.txt', 'w') as fLog:
```

Comment:
When opening files, I usually explicitly specify `b` or `t` in addition to the read/write mode to make sure (and communicate) that the resulting file object is in binary or in text mode.

Reply:
Seems like a good plan. I am also contemplating doing the file log differently (or abandoning it altogether) cause while `with open` is nice and pythonic, the empty file when no errors are found bothers me. And I guess in a simple script we should be able to properly deal with `.close()` and error handling :)